### PR TITLE
Simplify how user-provided quantifier attributes are handled

### PR DESCRIPTION
**Pull request by ajreynol**
This makes INST_ATTRIBUTE (optionally) take multiple children, giving a way for the user to set attributes on quantified formulas, which does not require a new API command.

This is a special case of term annotations that occur as the body of a quantified formula.

If we need to extend our API to support further user-provided attributes, we should use a similar approach, e.g. a new kind ANNOTATE.

### CODE REVIEW

**Review by nafur — APPROVED** [review]
LGTM, but I'd really like @aniemetz have a look as well.

File: src/theory/quantifiers/quantifiers_attributes.cpp [hunk 1]
```diff
@@ -36,7 +36,10 @@ bool QAttributes::isStandard() const
 
 QuantAttributes::QuantAttributes() {}
 
-void QuantAttributes::setUserAttribute( const std::string& attr, Node n, std::vector< Node >& node_values, std::string str_value ){
+void QuantAttributes::setUserAttribute(const std::string& attr,
+                                       Node n,
+                                       std::vector<Node>& node_values)
```
Comment (aniemetz) [inline src/theory/quantifiers/quantifiers_attributes.cpp:41]: `node_values` should be const.

File: src/theory/quantifiers/quantifiers_attributes.cpp [hunk 2]
```diff
@@ -200,7 +204,28 @@ void QuantAttributes::computeQuantAttributes( Node q, QAttributes& qa ){
...
+        // (INST_ATTRIBUTE v) where v has an internal attribute set on it.
+        // We distinguish these two cases by checking the kind of the first
+        // child.
+        if (q[2][i][0].getKind() == CONST_STRING)
+        {
+          // make a dummy variable to be used below
+          avar = nm->mkBoundVar(nm->booleanType());
+          std::vector<Node> nodeValues(q[2][i].begin() + 1, q[2][i].end());
+          // set user attribute on the dummy variable
+          setUserAttribute(
```
Comment (aniemetz) [inline src/theory/quantifiers/quantifiers_attributes.cpp:221]: How deep may this recurse in the worst case?

**Review by aniemetz — CHANGES_REQUESTED** [review]

File: src/theory/quantifiers/quantifiers_attributes.cpp [hunk 3]
```diff
@@ -36,7 +36,10 @@ bool QAttributes::isStandard() const
 
 QuantAttributes::QuantAttributes() {}
 
-void QuantAttributes::setUserAttribute( const std::string& attr, Node n, std::vector< Node >& node_values, std::string str_value ){
+void QuantAttributes::setUserAttribute(const std::string& attr,
+                                       Node n,
```
Comment (aniemetz) [inline src/theory/quantifiers/quantifiers_attributes.cpp:40]: Does this have to be `Node`? Why not a reference? Or a `TNode`?

File: src/theory/quantifiers/quantifiers_attributes.cpp (hunk 2 above)
Comment (ajreynol) [reply src/theory/quantifiers/quantifiers_attributes.cpp:221]: There is no recursion here, this is called from computeQuantAttributes.

Comment (aniemetz) [reply src/theory/quantifiers/quantifiers_attributes.cpp:220]: Ah right, I missed that this was collapsed in. All good :)

**Review by aniemetz — APPROVED** [review]

**Pull request merged**